CONTRIBUTING.md

```diff
@@ -24,19 +24,79 @@ Please follow these steps when making pull requests (PRs):
     Please give the PR a descriptive name.
 
 1.  As you make progress on your PR, keep your branch up-to-date with the `master` branch which
-        may have been updated after starting your PR.
+        may have been updated *after* starting your PR.
     Your PR **MUST** be updated to reflect changes to `master` in order to be merged.
-    Use the following procedure for updating your branch:
+    Use the following procedure for updating your branch and are ready to commit your changes:
 
     <!--- language: lang-shell -->
 
-        # While on your PR branch <issue-branch>
+        ## While on your PR branch <issue-branch> hosted at <your-remote> repository:
         git commit
         git checkout master
-        git pull origin --rebase
+        git pull <illixr-remote> --rebase                           ## Line A
         git checkout <issue-branch>
-        git rebase master
-        git push origin <issue-branch> --force-with-lease
+        git rebase master                                           ## Line B
+        git push <your-remote> <issue-branch> --force-with-lease    ## Line C
+
+    If you want to update your branch while you are still making changes to your branch:
```

Comment:
One could merge these two sections with:

```
git commit # or git stash
...
# if you did git stash, then do: git stash pop && git commit
...
```

IMO easier to read and understand the differences/commonalities between the two occasions.